### PR TITLE
[Inductor UT] Reuse inductor UT for intel GPU `test/inductor/test_compiled_optimizers.py`

### PR DESCRIPTION
**Pull request by hoshibara**
[Inductor UT] Reuse Inductor test case for Intel GPU.
Reuse `test/inductor/test_compiled_optimizers.py`

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by EikanWang — COMMENTED** [review]
LGTM

File: torch/testing/_internal/common_optimizers.py [hunk 1]
```diff
@@ -1968,6 +1969,9 @@ def _get_optim_inputs_including_global_cliquey_kwargs(
                 skipIfMps,  # SparseAdam does not support MPS
                 "TestOptimRenewed",
             ),
+            DecorateInfo(
+                skipIfXpu(msg="SparseAdam doesn't currently work on the XPU stack"),  # SparseAdam does not support XPU
```
Comment (etaf) [inline torch/testing/_internal/common_optimizers.py:1973]: May I know why XPU don't support SparseAdam ?

**Review by etaf — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Review by mlazos — APPROVED** [review]

**Pull request closed without merging**